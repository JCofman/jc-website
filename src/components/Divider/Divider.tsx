import React from 'react';

import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledG = styled.g`
  fill: var(--color-text);
`;

const StyledDivider = styled.svg`
  display: block;
  width: 55px;
  height: 12px;
  margin: 0 auto;
`;

const Divider = (props) => (
  <StyledDivider>
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd" {...props}>
      <StyledG transform="translate(-483.000000, -1059.000000)">
        <g transform="translate(182.000000, 936.000000)">
          <g transform="translate(301.000000, 123.000000)">
            <rect fillOpacity="0.3" x="0" y="0" width="35" height="12" />
            <rect fillOpacity="0.1" x="20" y="0" width="35" height="12" />
          </g>
        </g>
      </StyledG>
    </g>
  </StyledDivider>
);

PropTypes.displayName = `Divider`;

export default Divider;
