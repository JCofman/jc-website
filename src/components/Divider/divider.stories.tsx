import React from 'react';
import { storiesOf } from '@storybook/react';
import Divider from './Divider';

storiesOf(`Divider`, module).add(`Basic`, () => {
  return <Divider />;
});
