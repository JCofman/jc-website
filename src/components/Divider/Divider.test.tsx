import React from 'react';
import { render } from '@testing-library/react';

import Divider from './Divider';

test(`Divider is rendered properly`, () => {
  const { container } = render(<Divider />);
  expect(container).toMatchSnapshot();
});
