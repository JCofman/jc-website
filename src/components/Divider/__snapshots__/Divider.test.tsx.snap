// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider is rendered properly 1`] = `
.c1 {
  fill: var(--color-text);
}

.c0 {
  display: block;
  width: 55px;
  height: 12px;
  margin: 0 auto;
}

<div>
  <svg
    class="c0"
  >
    <g
      fill="none"
      fill-rule="evenodd"
      stroke="none"
      stroke-width="1"
    >
      <g
        class="c1"
        transform="translate(-483.000000, -1059.000000)"
      >
        <g
          transform="translate(182.000000, 936.000000)"
        >
          <g
            transform="translate(301.000000, 123.000000)"
          >
            <rect
              fill-opacity="0.3"
              height="12"
              width="35"
              x="0"
              y="0"
            />
            <rect
              fill-opacity="0.1"
              height="12"
              width="35"
              x="20"
              y="0"
            />
          </g>
        </g>
      </g>
    </g>
  </svg>
</div>
`;
